src/main/java/core/basesyntax/fileservice/FileServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.fileservice;
+
+import core.basesyntax.validator.Validator;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FileServiceImpl implements FileService {
+    private static final int LINE_WITH_TITLES = 1;
+
+    @Override
+    public List<String[]> readFromFile(Path fileFrom) {
+        try {
+            return Files.lines(fileFrom)
+                    .skip(LINE_WITH_TITLES)
+                    .map(l -> l.split(","))
+                    .filter(new Validator())
```

Comment:
You validator screens out data from file and does not inform client about that. is it what you want?

Reply:
Validator here is the Predicate function. If will be incorrect record it throw exception